From 3db9861d855460ff96fda993682ed14bedf1ed17 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Thu, 9 May 2013 13:14:59 +0800
Subject: [PATCH 321/334] Change the internal SRAM memory type
 MT_MEMORY_NONCACHED

Because MT_DEVICE is not executable in armv7, we change
the the internal SRAM memory type to MT_MEMORY_SO as commit
5fe012d4aee98ef82df66e6763882c735902ed0f,
but MT_MEMORY_SO is strongly ordered, this feature is not required
for the internal SRAM memory.

So we change the internal SRAM memory type to MT_MEMORY_NONCACHED.
As it seems that caching this internal SRAM memory is not necessary,
we chose the this memory type.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 arch/arm/mach-at91/setup.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-at91/setup.c b/arch/arm/mach-at91/setup.c
index 2c416cf..98d55ef 100644
--- a/arch/arm/mach-at91/setup.c
+++ b/arch/arm/mach-at91/setup.c
@@ -84,7 +84,7 @@ void __init at91_init_sram(int bank, unsigned long base, unsigned int length)
 
 	desc->pfn = __phys_to_pfn(base);
 	desc->length = length;
-	desc->type = MT_MEMORY_SO;
+	desc->type = MT_MEMORY_NONCACHED;
 
 	pr_info("AT91: sram at 0x%lx of 0x%x mapped at 0x%lx\n",
 		base, length, desc->virtual);
-- 
1.7.10.4

